core/src/main/java/org/kiwix/kiwixmobile/core/extensions/HeaderizableList.kt

```diff
@@ -0,0 +1,21 @@
+package org.kiwix.kiwixmobile.core.extensions
+
+inline class HeaderizableList<T>(val list: List<T>) {
```

Comment:
It looks like we have sacrificed some type information here, ITEM&HEADER was supposed to have an upperbound of SUPERTYPE and the resultant list was of type SUPERTYPE.

Generics can get funky but try to replicate that

Reply:
Hmm, Yes I am not very happy with this resulting class. I also had to cast all the list items to <ITEM> in the function calls which seems like poor style. I'll look into this function/class and its calls tomorrow. Is there a way to use the inline class as the class it is 'extended' from? I.e. headerizableList.size(). Currently I have to do: headerizableList.list.size().

Reply:
You could add methods on HeaderizableList but it sounds like a big fuss, really we only want to use this in 1 scenario and that is when adding headers so ideally the code will just be
```
val listWithHeaders = HeaderizableList(myList).foldOverAddingHeaders(/*blah blah*/)
```
where `listWithHeaders` is just a regular list

Reply:
Ok, good. Because that is how I have done it!